src/edge/models/genome.py

```diff
@@ -180,6 +180,71 @@ def changed_locations_by_fragment(self):
         changes = {Fragment.objects.get(pk=f): v for f, v in changes.items()}
         return changes
 
+    def get_coordinate_diff_from_parent_genome(self, p_genome_id):
+        # Get fragments contained by child and not by parent
+        p_fragment_ids = Fragment.objects.filter(genome=p_genome_id).values_list('id', flat=True)
+        child_only_fragments = self.fragments.filter().exclude(genome=p_genome_id)
+
+        # Relate child fragments to parent fragments
+        child_to_parent_f_ids = {}
+        for f in child_only_fragments:
+            curr_f = f
+            while curr_f is not None and curr_f.id not in p_fragment_ids:
+                curr_f = curr_f.parent
+            if curr_f is not None:
+                child_to_parent_f_ids[f.id] = curr_f.id
+        if len(child_to_parent_f_ids) == 0:
+            raise Exception("Genome input is not a parent by lineage")
+
+        # Get indexed child and parent fragments
+        child_to_parent_fs = {
+            Fragment.objects.get(pk=child_f_id).indexed_fragment():
+                Fragment.objects.get(pk=parent_f_id).indexed_fragment()
+            for child_f_id, parent_f_id in child_to_parent_f_ids.items()
+        }
+
+        # Get start to end chunks by fragment of diff regions between parent and child
+        fragment_to_starts_and_ends = {}
+        for child_f, parent_f in child_to_parent_fs.items():
+            print(child_f, parent_f)
+            parent_chunks, child_chunks = list(parent_f.chunks()), list(child_f.chunks())
+            starts_and_ends, current_start = {}, None
+            for child_i, c in enumerate(child_chunks):
+                if c in parent_chunks:
+                    parent_i = parent_chunks.index(c)
```

Comment:
I may be wrong -- does this .index here actually work? Because `Fragment.chunks()` returns objects. A chunk could be in parent and child, but `parent_f.chunks()` and `child_f.chunks()` may return different objects representing the same chunk.

Reply:
It does seem to work, as object comparison will just be looking for matching id/pk. Would there be different objects denoting the same chunk? I thought that any sections of discordant chunks between parent and child denote an operation, since any matching chunks should just be preserved from the parent.

Reply:
They are the same "chunk" by ID and content, but I believe separate calls to the Django ORM will return different in memory objects representing those chunks. I.e.

```
c = Chunk.objects.get(id=5)
d = Chunk.objects.get(id=5)
c.note = "hi"
print(d.note) # will not print "hi"
```

But, it's possible that they created the right Python machinery to handle `in` operator, by doing equivalence check via the pk/id field.

Reply:
The following simple test seems to pass, so I think your logic is correct

```
    def test_object_equivalence(self):
        a = Genome(name="foo")
        a.save()
        b = Genome.objects.get(id=a.id)
        c = Genome.objects.get(id=a.id)
        self.assertEquals(b in [c], True)
        b.name = "bar"
        self.assertNotEqual(b.name, c.name)
```